server/routes/api/helpscout.ts

```diff
@@ -0,0 +1,126 @@
+import * as express from "express";
+import {isHelpScoutIntegrationEnabled, validateHelpScoutSignature} from "../../middleware";
+import {IFormItem, IUser, User} from "../../schema";
+import bodyParser = require("body-parser");
+import * as moment from "moment-timezone";
+
+export const helpScoutRoutes = express.Router({"mergeParams": true});
+
+export type RequestWithRawBody = express.Request & { rawBody: string };
+
+helpScoutRoutes.route("/userInfo").post(
+	isHelpScoutIntegrationEnabled,
+	bodyParser.json({
+		verify: (req: RequestWithRawBody, res, buffer, encoding) => {
+			if (buffer && buffer.length) {
+				req.rawBody = buffer.toString(encoding || 'utf-8');
+			}
+		}
+	}),
+	validateHelpScoutSignature,
+	helpScoutUserInfoHandler
+);
+
+async function findUserByEmail(email: string) {
+	return User.findOne({
+		email
+	});
+}
+
+function findApplicationQuestion(applicationData: IFormItem[], questionName: string) {
+	const result = applicationData.find((item: IFormItem) => item.name === questionName);
+	return result ? result.value : "";
+}
+
+function createListBlock(title: string, content: string | string[] | Express.Multer.File | null) {
+	return `<li class="c-sb-list-item">
+			<span class="c-sb-list-item__label">${title}<span class="c-sb-list-item__text">${content}</span>
+		</span>
+	</li>`;
+}
+
+function badge(type: string, text: string) {
+	return `<span class="badge ${type}">${text}</span>`;
+}
+
+async function helpScoutUserInfoHandler(request: express.Request, response: express.Response) {
```

Comment:
This is a scary function because if the auth checks pass it can query information about any user. For purposes of monitoring it (and also improving general data privacy posture) I'd consider logging accesses somewhere and having it notify y'all when it is used.

Reply:
That's a good point.  Would it be worthwhile to directly validate the signature in the function vs. middleware to prevent it from accidentally being open to the world?  I wonder if we could also restrict it to be called by Help Scout IP addresses 🤔